src/components/material/material-buttons/online/MaterialButtonOnlinePublizon.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { FC } from "react";
+import { Manifestation } from "../../../../core/utils/types/entities";
+import {
+  hasPlayerManifestation,
+  hasReaderManifestation
+} from "../../../reader-player/helper";
+import MaterialSecondaryLink from "../generic/MaterialSecondaryLink";
+import MaterialSecondaryButton from "../generic/MaterialSecondaryButton";
+import { playerModalId } from "../../player-modal/helper";
+import { getManifestationIsbn } from "../../../../apps/material/helper";
+import { useModalButtonHandler } from "../../../../core/utils/modal";
+import { useText } from "../../../../core/utils/text";
+import { ButtonSize } from "../../../../core/utils/types/button";
+
+type MaterialButtonOnlinePublizonType = {
+  size?: ButtonSize;
+  manifestations: Manifestation[];
+  dataCy?: string;
+};
+
+const MaterialButtonOnlinePublizon: FC<MaterialButtonOnlinePublizonType> = ({
+  size,
+  manifestations,
+  dataCy = "material-button-online-publizon"
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+
+  return (
+    <>
+      {/* Todo: add logic for the reservation / loan / Read / Listen buttons here */}
+
+      {hasReaderManifestation(manifestations) && (
+        <MaterialSecondaryLink
+          label={t("onlineMaterialTeaserText", {
+            placeholders: { "@materialType": t("ebookText") }
+          })}
+          size={size || "large"}
+          url={
+            new URL(
+              `/reader?identifier=${getManifestationIsbn(manifestations[0])}`,
+              window.location.href
+            )
+          }
+          dataCy={`${dataCy}-reader-teaser`}
+        />
+      )}
+      {hasPlayerManifestation(manifestations) && (
+        <MaterialSecondaryButton
+          label={t("onlineMaterialTeaserText", {
+            placeholders: { "@materialType": t("audiobookText") }
+          })}
+          size={size || "large"}
+          onClick={() => {
+            open(playerModalId(getManifestationIsbn(manifestations[0])));
+          }}
+          dataCy={`${dataCy}-player-teaser`}
+          ariaDescribedBy={t("onlineMaterialTeaserText")}
+        />
+      )}
```

Comment:
Maybe we should add some sort of backup if `hasReaderManifestation` is false-y and `hasPlayerManifestation` is also false'y. 
Possibly just some text that says we can't show the material - like "missing material" or "missing data", or a disabled button that doesn't do anything, just has a text inside that convey this message.

Reply:
I like the idea. 😊 When we implement the rest of the logic for loans and reservations, we need to consider several things such as quotas, quantities, loans, etc. And here, I definitely think we should disable the button and display an explanatory text underneath. This logic will be implanted when we get there, and there are already tickets for it.

As it is right now, users will always be able to try the teaser for the materials that has the right type.